factory_reset/factory_reset.proto

```diff
@@ -34,8 +37,20 @@ message StartRequest {
   bool factory_os = 1;
   // Instructs the Target to zero fill persistent storage state data.
   bool zero_fill = 2;
-  // Instructs the Target to retain certificates
+  // Instructs the Target to retain certificates.
   bool retain_certs = 3;
+  // Instructs the Target to populate with the provided Ip/Routing information
+  // and skip DHCP boot.
+  ManagementConfig mgmt_config = 4;
+}
+
+message ManagementConfig {
```

Comment:
Would it make more sense to use https://github.com/openconfig/bootz/blob/main/proto/bootz.proto#L225-L236 here?  The idea being:

- factory reset the config
- apply the attached config. (which might include disabling dhcp and anything else you want to override)
- reboot

Reply:
The problem is that this is not a traditional config.

This is a special config, that should be consumed at the start of the bootz process (which is typically invoked before any startup/running-config is involved) and by the bootz daemon only (it has no meaning to the other system components).

Reply:
The bootconfig message contains a parameter `bootloader_config` which is intended to contain items that are invoked/processed before any startup/running config is involved.  If an implementation requires `bootloader_config` to contain some data to enable/disable the DHCP client used during bootz, that is fine.  If the implementation uses OC config to do this, that is also fine.  

If my comment is not on-track with your concerns, maybe you can share more details about the implementation so we can try and map it to bootz.  Happy to do this offline with you if necessary and then we can follow up with some proposal here.

```
message BootConfig {
  // Proprietary key-value parameters that are required as part of boot
  // configuration (e.g., feature flags, or vendor-specific hardware knobs).
  google.protobuf.Struct metadata = 1;
  // Native format vendor configuration.
  bytes vendor_config = 2;
  // JSON rendered OC configuration.
  bytes oc_config = 3;
  // Bootloader key-value parameters that are required as part of boot
  // configuration.
  google.protobuf.Struct bootloader_config = 4;
}
```

Reply:
@LimeHat let me know what you think.  Would like to keep this PR moving.

Reply:
> If an implementation requires bootloader_config to contain some data to enable/disable the DHCP client used during bootz, that is fine.  If the implementation uses OC config to do this, that is also fine.

There's a bit more to this proposal than a simple enablement/disablement of DHCP.
The core part of the bootz protocol is a DHCP response. It includes not only information about IP config, but also the bootstrap URL(s).

This proposal, in essence, is a way to substitute a real DHCP response with a pre-defined static message received via gNOI (pseudo-DHCP).

There are different ways to store this data. It can be stored in bootloader parameters (as you indicated, but this will be more than a flag), in can be stored in persistent memory (e.g. as a file or another implementation-dependent data structure). Probably it can't be stored as OpenConfig data (for instance, I don't think you can specify bootstrap URLs; and if the author wants to preserve compatibility with `FactoryReset`, it cannot be stored as a regular bootstrap cfg which is supposed to be erased during the reset). As a user, you probably don't to be concerned with implementation-specific details in order to use this.

I also don't want to couple this data with `BootConfig` message, for a number of reasons, including
- in my opinion, these functions are logically independent, and,
- overloading this structure will require the user to know and submit unneeded data to simply enable this pseudo-dhcp thing (e.g. the `bytes vendor_config` field will have to be present),
- the whole `SetBootConfigRequest` RPC is designed to replace the data that was received from a bootstrap server, and that data only.

Personally, I still think that a new service is the best approach. If you'd prefer to extend the existing `BootConfig` service, I would suggest to create a new RPC for this feature, that seems like a reasonable middle ground. 

Feel free to ping me in chat if you want to discuss further offline.